testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy

```diff
@@ -167,7 +249,11 @@ abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
         def coreLibs = contentsDir.file("lib").listFiles().findAll {
             it.name.startsWith("gradle-") && !it.name.startsWith("gradle-api-metadata") && !it.name.startsWith("gradle-kotlin-dsl")
         }
+
+        coreLibs.each { assert coreLibsContains(it.name) }
         assert coreLibs.size() == coreLibJarsCount
+        assert coreLibs.size() == coreLibsFileNames.size()
```

Comment:
💭  I think we can improve DX here for when people need to update things.
Now that we have an explicit list, we can provide more actionable advice on assertion failure.

Specifically, we can now check what are some jars in the `coreLibs` that are not in the list. And we also can check what are some jars that are in the list, but not in the `coreLibs`. The expected result is zero in both cases. But the benefit is that one check can give the developers an immediate answer about all the changes that need to happen, as opposed to incrementally failing in the `each` call.

So, I would do it like this:
```groovy
def expectedCoreLibs = coreLibsFileNames.toSet()
def actualCoreLibs = coreLibs.collect { dropVersionSuffix(it) }.toSet()

def unknownCoreLibs = (actualCoreLibs - expectedCoreLibs).sorted()
def missingCoreLibs = (expectedCoreLibs - actualCoreLibs).sorted()

verifyAll {
    unknownCoreLibs.isEmpty()
    missingCoreLibs.isEmpty()
}
```

We can remove the checks for sizes, because they are not informative but also redundant with the suggested changes.

The benefit of this approach is that no matter what sort of refactoring has been done to the modules (new modules added / removed / renamed), there is a single assertion error that explains everything that needs to be changed, and the change can be done in one iteration.